pkg/apis/kops/validation/validation.go

```diff
@@ -600,6 +595,30 @@ func validateContainerRuntime(runtime *string, fldPath *field.Path) field.ErrorL
 	return allErrs
 }
 
+func validateDockerConfig(config *kops.DockerConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if config.Version != nil {
+		if strings.HasPrefix(*config.Version, "1.1") {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("version"), config.Version,
+				"version is no longer available: https://www.docker.com/blog/changes-dockerproject-org-apt-yum-repositories/"))
+		} else {
+			valid := []string{"17.03.2", "17.09.0", "18.03.1", "18.06.1", "18.06.2", "18.06.3", "18.09.3", "18.09.9", "19.03.4", "19.03.8"}
```

Comment:
What used to happen if a user specified a version that was not in this list? like [18.09.8](https://docs.docker.com/engine/release-notes/18.09/#18098) for example.

Reply:
Any Docker version would have been accepted and NodeUp would fail if it was not listed.
It can still fail as not all versions have support for all distros.

Reply:
hm does that mean that if a specific version of docker was already installed in the AMI and a user specified a version not in this list, they used to have a functioning cluster with their docker version but now their `update cluster` command would fail? I'm just trying to think through any concerns with adding this validation...

Reply:
If a specific Docker version is install on a AMI, `skipInstall` would have been used and the Docker version would not matter much, I guess.